src/Illuminate/Contracts/Database/Eloquent/Hydratable.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Illuminate\Contracts\Database\Eloquent;
+
+interface Hydratable
+{
+    /**
+     * Get the value of the entity's primary key.
+     *
+     * @return mixed
```

Comment:
Why would the key be anything other than a string? Maybe I'm missing something. The description of this function is a bit odd, it's called `getKey`, but the description is more like `getValue`?

Reply:
I might be wrong, but I guess because it could be an object. For example YouTrack Issue identifier is composition of 2 identifiers: `{$projectName}-{$issueAutoIncrementId}`.

```php
class IssueId
{
    private $id;
    private $projectName;

    public function __construct(string $projectName, int $id)
    {
        $this->projectName = $projectName;
        $this->id = $id;
    }

    public function __toString(): string
    {
        return $this->projectName . '-' . $this->id;
    }
}
```

Reply:
Maybe we should force cast it to a string?

Reply:
IMO `mixed` is a good choice

Reply:
Oh, it is supposed to be an integer or a string. I just missed that detail in the doc block. I will fix it. Basically, it is the same as `$model->getKey()`. It is used to match results to the parent relation.

I think we might be able to remove `getKey()` from the interface and rely on `getAttribute()`. I would have to refactor some of the relations.

Take a look at how MorphTo matches results... https://github.com/laravel/framework/blob/5.6/src/Illuminate/Database/Eloquent/Relations/MorphTo.php#L183

I just have to work this a bit more.